In [2]:
import os
import time
import requests
import pandas as pd
from datetime import date
from dotenv import load_dotenv

load_dotenv() 
print ("Libraries imported")

Libraries imported


In [3]:
API_KEY = os.getenv("LEGI_API_KEY")

In [ ]:
url = "https://api.legiscan.com/"
## I need to get a access code from this request used to call the GetDataset url.

In [10]:
params = {
    "key": API_KEY,
    "op": "getDatasetList",
    "state": "US"
}

response = requests.get(url, params=params)

print(response.status_code)

data = response.json()
print(data.keys())


200
dict_keys(['status', 'datasetlist'])


In [11]:
status = data["status"]
print(status)

OK


In [15]:
print(data["datasetlist"][0])


{'state_id': 52, 'session_id': 2199, 'year_start': 2025, 'year_end': 2026, 'prefile': 0, 'sine_die': 0, 'prior': 0, 'special': 0, 'session_tag': 'Regular Session', 'session_title': '2025-2026 Regular Session', 'session_name': '119th Congress', 'dataset_date': '2026-09-06', 'dataset_hash': '99dcea213d48715d5dc603e1a48da9ac', 'dataset_size': 62930520, 'dataset_size_csv': 4403830, 'access_key': '7GNHYv7EcN1RPXURARflAn'}


9/11/2026: <br>
    session_id: 2199 <br>
    access_key: '7GNHYv7EcN1RPXURARflAn' <br>
    dataset_hash: '99dcea213d48715d5dc603e1a48da9ac' <br>